exercises/risk-risiko.cpp

```diff
@@ -27,3 +27,102 @@
   M 3 vs 3 => blue win
   O 2 vs 1 => red win
 */
+#include <iostream>
+using namespace std;
+class Dice
+{
+public:
+  int value;
+  Dice(int n)
+  {
+    value = (rand() % n) + 1;
+  }
+};
+
+int current_time_nanoseconds()
+{
+  struct timespec tm;
+  clock_gettime(CLOCK_REALTIME, &tm);
+  return tm.tv_nsec;
+}
+
+int *getMax3(Dice *dice_array)
+{
+  int max1, max2, max3;
+  max1 = max2 = max3 = 0;
+
+  int k = 0;
+  for (int i = 0; i < 6; i++)
+  {
+    if (max1 < dice_array[i].value)
+    {
+      max1 = dice_array[i].value;
+      k = i;
+    }
+  }
+  dice_array[k].value = 0;
+  k = 0;
+
+  for (int i = 0; i < 6; i++)
+  {
+    if (max2 < dice_array[i].value)
+    {
+      max2 = dice_array[i].value;
+      k = i;
+    }
+  }
+  dice_array[k].value = 0;
+
+  k = 0;
+  for (int i = 0; i < 6; i++)
+  {
+    if (max3 < dice_array[i].value)
+    {
+      max3 = dice_array[i].value;
+      k = i;
+    }
+  }
```

Comment:
sbaglio o hai fatto 3 cicli for uguali? non sarebbe meglio creare una funzione che eviti la ridondanza di codice?

Reply:
refactor effettuato, messo altre funzioni e inserite dentro un ciclo

Reply:
very very nice 👍